test/unit/host_aspect_configuration_test.rb

```diff
@@ -0,0 +1,47 @@
+require 'test_helper'
+
+class HostAspectConfigurationTest < ActiveSupport::TestCase
+  class MyCoolClass
+  end
+
+  test 'gives readonly access to the registry' do
+    config = HostAspects::Configuration.new
+    config.register 'MyCoolAspect', :my_cool_model
+    hash = config.registered_aspects
```

Comment:
I'm not sure I follow why saving `registered_aspects` in a variable then modifying it and checking the original `registered_aspects` tests `registered_aspects` is read only? That'd be a valid test in a pass-by-reference language but Ruby is pass by value. Would this work if this is what you're trying to test?? 

```ruby
assert_difference('config.registered_aspects.count', 0) do
  config.registered_aspects.delete(:MyCoolAspect)
end
```